### PR TITLE
Add missing default recurring change rights

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Caught by end to end tests (Cypress) during prototyping.

### CODE REVIEW

File: install/empty_data.php [hunk 1]
```diff
@@ -7717,6 +7717,31 @@ public function getEmptyData(): array
...
             ],
+            [
+                'profiles_id' => self::PROFILE_SUPERVISOR,
+                'name' => 'recurrentchange',
+                'rights' => ALLSTANDARDRIGHT,
+            ],
+            [
+                'profiles_id' => self::PROFILE_HOTLINER,
+                'name' => 'recurrentchange',
+                'rights' => ALLSTANDARDRIGHT,
```
Comment (cedric-anne) [inline install/empty_data.php:7728]:
```suggestion
                'rights' => self::RIGHT_NONE,
```

File: install/empty_data.php [hunk 2]
```diff
@@ -7717,6 +7717,31 @@ public function getEmptyData(): array
...
+            ],
+            [
+                'profiles_id' => self::PROFILE_TECHNICIAN,
+                'name' => 'recurrentchange',
+                'rights' => ALLSTANDARDRIGHT,
```
Comment (cedric-anne) [inline install/empty_data.php:7733]:
```suggestion
                'rights' => READ,
```

File: install/empty_data.php [hunk 3]
```diff
@@ -7717,6 +7717,31 @@ public function getEmptyData(): array
                 'name' => 'database',
                 'rights' => READ,
             ],
+            [
+                'profiles_id' => self::PROFILE_SUPERVISOR,
+                'name' => 'recurrentchange',
+                'rights' => ALLSTANDARDRIGHT,
```
Comment (cedric-anne) [inline install/empty_data.php:7723]:
```suggestion
                'rights' => READ,
```

File: install/empty_data.php [hunk 4]
```diff
@@ -7717,6 +7717,31 @@ public function getEmptyData(): array
...
+            ],
+            [
+                'profiles_id' => self::PROFILE_TECHNICIAN,
+                'name' => 'recurrentchange',
+                'rights' => ALLSTANDARDRIGHT,
+            ],
+            [
+                'profiles_id' => self::PROFILE_ADMIN,
+                'name' => 'recurrentchange',
+                'rights' => ALLSTANDARDRIGHT,
```
Comment (cedric-anne) [inline install/empty_data.php:7738]:
```suggestion
                'rights' => READ,
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]
I guess we should use same default rights as for `ticketrecurrent`.
Also, the `readonly` profile should have the `READ` right.

**Comment by cconard96** [comment]
The rights given here match what is given from the migration.
If a profile has the Update right for Changes, they are given all standard rights for recurring changes.

Are you saying the rights given during the migration should be different as well?

**Comment by cedric-anne** [comment]
> The rights given here match what is given from the migration. If a profile has the Update right for Changes, they are given all standard rights for recurring changes.
> 
> Are you saying the rights given during the migration should be different as well?

I am not sure of what should be done in migration.

**Comment by cedric-anne** [comment]
> recurrentchange

I take a look, and problem is that it may be complex to define what is the best right to apply during migration. IMHO we should not give right to create recurring changes for users that only have the right to update a change. We should probably give these rights only to super administrators (i.e. using the default `['config' => READ | UPDATE]` requirement).

@orthagh your opinion ?

**Comment by orthagh** [comment]
If we can matches profile with update on changes and ticketreccurent, i think we can apply them the new right.
Otherwise i agree, config update should be the safe way

**Review by cedric-anne — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]
I added missing profiles, even if they have `self::RIGHT_NONE`, as this is what is done on `ticketrecurrent` right in this file.

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

File: install/migrations/update_9.5.x_to_10.0.0/recurrentchange.php [hunk 5]
```diff
@@ -83,5 +83,6 @@
 }
 
 $migration->addRight('recurrentchange', ALLSTANDARDRIGHT, [
-    'change' => UPDATE
+    'change' => UPDATE,
```
Comment (trasher) [inline install/migrations/update_9.5.x_to_10.0.0/recurrentchange.php:86]: should not it be added in 10.0.2 migration?

Comment (cedric-anne) [reply install/migrations/update_9.5.x_to_10.0.0/recurrentchange.php:86]: No, we cannot "downgrade" rights once they have been provided, as people may already have reviewed them.

**Pull request merged**